units/src/amount/signed.rs

```diff
@@ -55,20 +55,65 @@ impl SignedAmount {
     pub const ONE_SAT: Self = SignedAmount(1);
     /// Exactly one bitcoin.
     pub const ONE_BTC: Self = SignedAmount(100_000_000);
-    /// The maximum value allowed as an amount. Useful for sanity checking.
+    /// The maximum value allowed as an amount.
     pub const MAX_MONEY: Self = SignedAmount(21_000_000 * 100_000_000);
     /// The minimum value of an amount.
     pub const MIN: Self = SignedAmount(-21_000_000 * 100_000_000);
     /// The maximum value of an amount.
     pub const MAX: Self = SignedAmount::MAX_MONEY;
 
     /// Constructs a new [`SignedAmount`] with satoshi precision and the given number of satoshis.
-    pub const fn from_sat(satoshi: i64) -> SignedAmount { SignedAmount(satoshi) }
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use bitcoin_units::SignedAmount;
+    /// # let sat = -100_000;
+    /// let amount = SignedAmount::from_sat(sat).expect("sat is less than MAX_MONEY");
+    /// assert_eq!(amount.to_sat(), sat);
+    /// ```
+    pub const fn from_sat(satoshi: i64) -> Result<SignedAmount, OutOfRangeError> {
+        if satoshi < Self::MIN.to_sat() {
+            Err(OutOfRangeError { is_signed: true, is_greater_than_max: false })
+        } else if satoshi > Self::MAX_MONEY.to_sat() {
```

Comment:
Why is this change to `MAX_MONEY` here instead of `MAX`?  It seems like this was on purpose since I had used `MAX`: https://github.com/yancyribbens/rust-bitcoin/blob/5a428f175c08fef09817187f826adaf66044de1e/units/src/amount/signed.rs#L73